### PR TITLE
Support arm64 architecture for tests.

### PR DESCRIPTION
**Pull request by pho3nixf1re**
I have a couple failing tests on arm64. I am looking into them but overall this seems to be working great and should allow official builds for arm.

Supersedes pull requests:
- #353
- #292 

This will probably close:
- #458 
- #287
- #275

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,18 +6,20 @@
 #            - ubuntu-toolchain-r-test
 #        packages:
 #            - openvpn
-
-services:
-    - docker
+language: minimal
```
Comment (pho3nixf1re) [inline .travis.yml]: Travis assumes Ruby by default and thus flagged builds for installing rvm and a default ruby. Using `minimal` reduces that overhead while still providing basic utilities like Docker out of the box.

**Comment by pho3nixf1re** [comment]
@kylemanna my only failing tests are the following:

```
failed
	'dual-proto' [5/8]...the input device is not a TTY
failed
	'otp' [6/8]...the input device is not a TTY
failed
	'iptables' [7/8]...the input device is not a TTY
failed
	'revocation' [8/8]...the input device is not a TTY
failed
The command "test/run.sh "$image"" exited with 1.
```

I'm really not sure why it would have a problem with TTY? Could there be something missing in the base arm image from travis?

**Comment by pho3nixf1re** [comment]
I am also seeing this for the `amd64` builds:

```
passed
	'otp' [6/8]...
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#build-times-out-because-no-output-was-received
The build has been terminated
```

**Comment by pho3nixf1re** [comment]
That OTP test hanging is due to a bug in the EasyRSA lib. I am going to add the fix into my PR here as a discrete commit. Some other PR's here have the temporary workaround as well. Ultimately it'll JustWork™️ once `alpine` is upgraded.

**Comment by maurerle** [comment]
Are there any news regarding this issue?

**Comment by pho3nixf1re** [comment]
I would still like this to be completed but I ran into issues getting these tests to pass. The actual arm64 builds work just fine. I need to circle back on this and see if the underlying issues are now fixed.

**Comment by juchong** [comment]
Hi folks, any update on this?

**Comment by pho3nixf1re** [comment]
Unfortunately no. I decided to move to WireGuard as the existing containers for it worked out-of-the-box and supported ARM builds better. There just doesn't seem to be core interest in support ARM for this project (not a complaint, just a statement of observation). It would be good to leave this open for someone else to pick up.

**Comment by yywing** [comment]
How about this pr？ I want use this image on arm.

**Comment by odidev** [comment]
I have successfully built and tested the image for both amd64 and arm64 architecture and also pushed it through Travis-ci on dockerhub.

**Commit Link** - https://github.com/pheobe07/docker-openvpn/commit/b67f766f8abf791ae15e16153a09b16732749962

**Travis Link** - https://travis-ci.com/github/pheobe07/docker-openvpn/builds/209627748

**Dockerhub Link** - https://hub.docker.com/repository/docker/odidev/openvpn

I have used arm64-graviton which is added for travis-ci.com as per https://blog.travis-ci.com/2020-09-11-arm-on-aws. So to use arm64-graviton2 in travis we need to migrate this from **travis-ci.org** to **travis-ci.com**.

Please share your thoughts on this and it will be really helpful if arm64 image is released for openvpn.

**Comment by hppyworld** [comment]
> I have successfully built and tested the image for both amd64 and arm64 architecture and also pushed it through Travis-ci on dockerhub.
> 
> **Commit Link** - [pheobe07@b67f766](https://github.com/pheobe07/docker-openvpn/commit/b67f766f8abf791ae15e16153a09b16732749962)
> 
> **Travis Link** - https://travis-ci.com/github/pheobe07/docker-openvpn/builds/209627748
> 
> **Dockerhub Link** - https://hub.docker.com/repository/docker/odidev/openvpn
> 
> I have used arm64-graviton which is added for travis-ci.com as per https://blog.travis-ci.com/2020-09-11-arm-on-aws. So to use arm64-graviton2 in travis we need to migrate this from **travis-ci.org** to **travis-ci.com**.
> 
> Please share your thoughts on this and it will be really helpful if arm64 image is released for openvpn.

Any progress. Still its not merged. Can you please provide docker image for arm64 directly on docker hub. Thanks.

**Comment by odidev** [comment]
> 
> 
> I have successfully built and tested the image for both amd64 and arm64 architecture and also pushed it through Travis-ci on dockerhub.
> 
> **Commit Link** - [pheobe07@b67f766](https://github.com/pheobe07/docker-openvpn/commit/b67f766f8abf791ae15e16153a09b16732749962)
> 
> **Travis Link** - https://travis-ci.com/github/pheobe07/docker-openvpn/builds/209627748
> 
> **Dockerhub Link** - https://hub.docker.com/repository/docker/odidev/openvpn
> 
> I have used arm64-graviton which is added for travis-ci.com as per https://blog.travis-ci.com/2020-09-11-arm-on-aws. So to use arm64-graviton2 in travis we need to migrate this from **travis-ci.org** to **travis-ci.com**.
> 
> Please share your thoughts on this and it will be really helpful if arm64 image is released for openvpn.



> 
> 
> I have successfully built and tested the image for both amd64 and arm64 architecture and also pushed it through Travis-ci on dockerhub.
> 
> **Commit Link** - [pheobe07@b67f766](https://github.com/pheobe07/docker-openvpn/commit/b67f766f8abf791ae15e16153a09b16732749962)
> 
> **Travis Link** - https://travis-ci.com/github/pheobe07/docker-openvpn/builds/209627748
> 
> **Dockerhub Link** - https://hub.docker.com/repository/docker/odidev/openvpn
> 
> I have used arm64-graviton which is added for travis-ci.com as per https://blog.travis-ci.com/2020-09-11-arm-on-aws. So to use arm64-graviton2 in travis we need to migrate this from **travis-ci.org** to **travis-ci.com**.
> 
> Please share your thoughts on this and it will be really helpful if arm64 image is released for openvpn.

Can you please have a look at my changes? If the changes looks good I can raise a separate PR for releasing image for arm64. 

It will be really helpful if arm64 image is released. 

May I know when are you planning on releasing the images?

**Comment by odidev** [comment]
Can you please have a look at my changes? If the changes looks good I can raise a separate PR for releasing image for arm64. 

It will be really helpful if arm64 image is released. 

May I know when are you planning on releasing the images?

**Pull request closed without merging**